webui/react/src/components/BatchActionConfirmModal.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+import { ExperimentAction } from '../types';
+
+import { Modal } from './kit/Modal';
+
+interface Props {
+  batchAction: ExperimentAction;
+  itemName?: string;
+  selectAll?: boolean;
+  danger?: boolean;
+  onConfirm: () => Promise<void>;
+  onClose?: () => void;
+}
+
+export const CONFIRM_BUTTON_LABEL = 'Confirm';
+
+const BatchActionConfirmModalComponent: React.FC<Props> = ({
+  batchAction,
+  itemName = 'experiment',
+  selectAll,
+  onConfirm,
+  onClose,
+}: Props) => {
+  const danger = /(cancel|kill)/i.test(batchAction);
```

Comment:
nit: this should work, but it seems less easy to read than something like `[ExperimentAction.Cancel, ExperimentAction.Kill].includes(batchAction)`.

fix: Delete should also be marked as dangerous.

Reply:
great catch on this one too!